src/model/table.py

```diff
@@ -20,8 +27,74 @@ def set(self):
     def delete(self):
         pass
 
-    def get(self):
-        pass
+    def get(self, query):
+        efficient_index = self.__get_efficient_index__(query)
+        if efficient_index is None:
+            efficient_keys = self.__get_all_primary_keys__()
+        else:
+            efficient_keys = efficient_index.get_primary_keys(query[efficient_index.name])
+        found_objects = self.__get_rows__(efficient_keys)
+        return self.__filter_by_query__(found_objects, query)
 
-    def get_by_primary_key(self):
-        pass
+    def __get_efficient_index__(self, query):
+        if not query or str(query).isspace:
+            return None
+        if self.__table_metadata__.primary_key in query.keys():
+            return PrimaryKeyIndex(self.__table_metadata__.primary_key, self.__table_metadata__)
+        index_names = set(self.__table_metadata__.get_indices_names())
+        most_efficient = None
+        for item in query.keys():
+            if item in index_names:
+                current_index = self.__table_metadata__.get_index(item)
+                if not most_efficient or current_index.compare(most_efficient) == -1:
+                    most_efficient = current_index
+        return most_efficient
+
+    def __get_all_primary_keys__(self):
+        primary_keys = []
+        for primary_key_path in os.listdir(os.path.join(self.__path__, "data")):
+            primary_key = self.__get_primary_key_from_path__(primary_key_path)
+            if primary_key:
+                primary_keys.append(primary_key)
```

Comment:
You should here return the keys without `.json`

Reply:
It already removes it in __get_primary_key_from_path__ function.